.github/workflows/lockfiles.yaml

```diff
@@ -0,0 +1,409 @@
+            echo '  <source media="(prefers-color-scheme: dark)" srcset="stderr.log.github-dark.png">'
+            echo '  <source media="(prefers-color-scheme: light)" srcset="stderr.log.github.svg">'
+            echo '  <source media="(prefers-color-scheme: light)" srcset="stderr.log.github.png">'
+            echo '  <img alt="Terminal screenshot of lockfile diff in color. The text from the image is included below."'
+            echo '    src="stderr.log.github.png">'
+            echo '</picture>'
+            echo
+            echo '<details>'
+            echo '  <summary>Lockfile diff: ${{ env.LOCKFILE }} (plain text)</summary>'
+            echo
+            echo '```'
+            cat ${STDERR_LOG}.txt
+            echo '```'
+            echo
+          ) | tee -a ${STDERR_LOG}.md >> ${GITHUB_STEP_SUMMARY}
+          (
+            echo '[!NOTE]'
+            echo 'Diff base (git ref): *${{ steps.pr.outputs.PR_BASE_REF }}* '
+            echo
+          ) >> ${GITHUB_STEP_SUMMARY} # only add this note to the job summary
+          (
+            echo '</details>'
+            echo
+          ) | tee -a ${STDERR_LOG}.md >> ${GITHUB_STEP_SUMMARY}
+
+      - name: Upload lockfile and lockfile diff files
+        uses: actions/upload-artifact@v4
+        with:
+          name: lockfile-${{ matrix.resolve }}
+          path: ${{ env.OUTPUT_DIR }}
+        if: steps.lockfile.outputs.CHANGED == 'true'
+
+      - name: Upload pants log
+        uses: actions/upload-artifact@v4
+        with:
+          name: pants-log-${{ matrix.resolve }}
+          path: .pants.d/pants.log
+        if: always()  # We want the log even on failures.
+
+  commit:
+    name: Commit regenerated ${{ matrix.resolve }} lockfile(s)
+    needs: [resolves, pr, regenerate]
+    runs-on: ubuntu-22.04
+    env:
+      # preserve the order of resolves from input (eg pants.toml has st2 first, before tools)
+      RESOLVES: ${{ join(fromJSON(needs.resolves.ouptuts.JSON), ''' ''') }}
+      RESOLVES_CSV: ${{ join(fromJSON(needs.resolves.ouptuts.JSON), ', ') }}
+      LOCKFILES: ${{ needs.resolves.outputs.LOCKFILES }}
+      # see also the 'Setup env' step
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+        with:
+          # a test uses a submodule, and pants needs access to it to calculate deps.
+          submodules: 'true'
+          ref: ${{ needs.pr.outputs.CHECKOUT_REF }}
+
+      # This step exists because we cannot use the 'env' context in 'jobs.<job>.env.*'.
+      - name: Setup env
+        run: |
+          echo "COMMIT_MSG=${{ env.OUTPUT_BASE_DIR }}/commit_msg.txt" >> ${GITHUB_ENV}
+          echo "PR_COMMENT=${{ env.OUTPUT_BASE_DIR }}/pr_comment.md" >> ${GITHUB_ENV}
+
+      - name: Create branch for new PR
+        if: ${{ inputs.pr == 'new' }}
+        run: |
+          git config --local user.name "github-actions[bot]"
+          git config --local user.email "41898282+github-actions[bot]@users.noreply.github.com"
```

Comment:
We could use one of our bot accounts instead. I went with the github actions bot account for now.

Reply:
Using something else to create PRs would require some kind of PAT or other token. To minimize maintenance, I don't want to create a PAT. Eventually we could probably use a GitHub App to make the PRs show up as coming from our bot. But, I'm not going to spend time on that now.

For now, I'll use @st2stanley as the git author, and @github-actions[bot] as the git committer. PR creation and PR comments will also be done by @github-actions[bot].